sympy/geometry/point.py

```diff
@@ -406,24 +440,23 @@ def unit(self):
         return self / abs(self)
 
     @staticmethod
-    def affine_rank(*points):
+    def affine_rank(*args):
         """The affine rank of a set of points is the dimension
         of the smallest affine space containing all the points.
         For example, if the points lie on a line (and are not all
         the same) their affine rank is 1.  If the points lie on a plane
         but not a line, their affine rank is 2.  By convention, the empty
         set has affine rank -1."""
 
-        if len(points) == 0:
+        if len(args) == 0:
             return -1
-
         # make sure we're genuinely points
         # and translate every point to the origin
-        points = [Point(i) for i in points]
+        points = Point._normalize_dimension(*[Point(i) for i in args])
```

Comment:
I see that you don't need this since you use the ambient dimension information to raise an error, but I am including it here to make it more transparent and to preserve the identity of the origin, e.g. if the args are `Point(1, 2)` and `Point(1, 2, 3)` the error message references `(-1, -2)` since the error is raised during the shift-to-origin step.